src/MolSim.cpp

```diff
@@ -73,25 +288,46 @@ int main(int argc, char *argsv[]) {
 }
 
 void calculateF() {
-  std::list<Particle>::iterator iterator;
-  iterator = particles.begin();
+  std::array<double, 3> sigma = {.0,.0,.0};
+  for (auto &p : particles) {
+    p.setOldF(p.getF());
+    p.setF(sigma);
+  }
+
+  for (auto &pair : particles.getParticlePairs()) {
+    Particle &p1 = pair.first.get();
+    Particle &p2 = pair.second.get();
 
-  for (auto &p1 : particles) {
-    for (auto &p2 : particles) {
-      // @TODO: insert calculation of forces here!
+    std::array<double, 3> sub = subtractVector(p2.getX(), p1.getX());
+    double normCubed = pow(magnitude(sub),3);
+
+    //prevent division by 0
+    if (normCubed == 0) {
+      continue;
     }
+
+
+    std::array<double, 3> fij = multiply_constant_vector(multiply_constant_vector(sub, p1.getM()*p2.getM()),1/normCubed);
+
+    p1.setF(addVector(p1.getF(),fij));
+    p2.setF(subtractVector(p2.getF(),fij));
   }
+
+
+
 }
 
 void calculateX() {
   for (auto &p : particles) {
-    // @TODO: insert calculation of position updates here!
-  }
+    std::array<double, 3> newX = addVector(p.getX(), addVector( multiply_constant_vector(p.getV(),delta_t) , multiply_constant_vector(p.getOldF(),0.5*pow(delta_t,2)/p.getM())));
+    p.setX(newX);
+ }
 }
 
 void calculateV() {
```

Comment:
additionally, both position and velocity update can be part of a `TimeDescretization.h` class as a potential refactoring